File: frontend/src/types/order.ts
import type { Side } from "./enums/side";
import type { OrderStatus } from "./enums/orderStatus";

export type Order = {
  id: number;
  side: Side;
  price: number;
  quantity: number;
  remaining_quantity: number;
  status: OrderStatus;
  timestamp: string;
};

File: backend/models/order.py
from pydantic import BaseModel
from datetime import datetime
from enum import Enum


class Side(str, Enum):
    BUY = "BUY"
    SELL = "SELL"


class OrderStatus(str, Enum):
    OPEN = "OPEN"
    PARTIALLY_FILLED = "PARTIALLY_FILLED"
    FILLED = "FILLED"
    CANCELLED = "CANCELLED"


class Order(BaseModel):
    id: int
    side: Side
    price: float
    quantity: float
    remaining_quantity: float
    status: OrderStatus
    timestamp: datetime
